src/pages/index.js

```diff
@@ -0,0 +1,349 @@
+plus.src = plusSrc;
+// The modal close button always returns NULL and I can't figure out why
+// const whitex = document.querySelector('modal__close-button-white');
+// whitex.src = whiteXSrc;
+
+const closeButtons = document.querySelectorAll(".modal__close-button-logo");
+closeButtons.forEach(button => {
+  button.src = grayXSrc; // Assign the image source
+});
+
+closeDeleteModalButton.addEventListener("click", () => {
+  closeModal(deleteModal);
+})
+
+cancelDeleteButton.addEventListener("click", () => {
+  closeModal(deleteModal);
+})
+
+// Select the modal
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = previewModal.querySelector(".modal__image");
+const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
+const previewModalCloseButton = previewModal.querySelector(
+  ".modal__close-button_type_preview"
+);
+const avatarModal = document.querySelector("#avatar-modal");
+
+// avatar form elements
+const avatarSubmitButton = avatarModal.querySelector(".modal__submit-button");
+const avatarCardElement = avatarModal.querySelector(".modal__form");
+const addAvatarModalCloseButton = avatarModal.querySelector(
+  ".modal__close-button"
+);
+const avatarLinkInput = avatarModal.querySelector("#profile-avatar-input");
+
+api.getAppInfo()
+.then(([cards, userInfo]) => {
+  console.log("Fetched cards:", cards);
+  console.log("Number of cards:", cards.length);
+  cards.forEach((item) => {
+    const cardElement = getCardElement(item);
+    cardsList.prepend(cardElement);
+  });
+  profileNameElement.textContent = userInfo.name;
+  profileDescriptionElement.textContent = userInfo.about;
+  avatar.src = userInfo.avatar;
+})
+.catch(console.error);
+
+// function to handle closing the modal with escape key
+function handleEscapeKey(event) {
+  if (event.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    if (openedModal) {
+      closeModal(openedModal);
+    }
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown",handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown",handleEscapeKey);
+}
+
+
+//add listeners for closing the modal on overlay click
+function addOverlayClickListener(modal) {
+  modal.addEventListener("click", (event) => {
+    if (event.target === modal) {
+      closeModal(modal);
+    }
+  });
+}
+
+// select all modals
+const modals = document.querySelectorAll(".modal");
+
+avatarModalButton.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+avatarCardElement.addEventListener("submit", handleAvatarSubmit);
+
+// add listener to each modal
+modals.forEach((modal) => {
+  addOverlayClickListener(modal);
+})
+
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **resetting** and **catching errors**


```js
// define a function for changing the button text. It accepts 4 params (the 2 last are optional with default texts)
export function renderLoading(isLoading, button, buttonText='Save', loadingText='Saving...') {
  if (isLoading) {
    button.textContent = loadingText
  } else {
    button.textContent = buttonText
  }
}

// define a universal function that accepts a request function, event and a default loading text 
function handleSubmit(request, evt, loadingText = 'Saving...') {
 // You need to prevent the default action in any submit handler
  evt.preventDefault();

  // the button is always available inside `event` as `submitter`
  const submitButton = evt.submitter;
  // fix the initial button text
  const initialText = submitButton.textContent;
  // change the button text before requesting
  renderLoading(true, submitButton, initialText, loadingText);
  // call the request function to be able to use the promise chain
  request()
    .then(() => {
      // any form should be reset after a successful response 
      // evt.target is the form in any submit handler		
      evt.target.reset();
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
    .catch(console.error)      

    // and in finally we need to stop loading
    .finally(() => {
      renderLoading(false, submitButton, initialText);
    });
}

```

Here is an example of handling the profile form submit:

```js
function handleProfileFormSubmit(evt) {
  // create a request function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally`
    return editProfile(nameInput.value, jobInput.value).then((userData) => {
      userName.textContent = userData.name;
      userJob.textContent = userData.about;
    });
  }
  // here we call handleSubmit passing the request and event (if you want a different loading text then you need to pass the 3rd param)
  handleSubmit(makeRequest, evt);
}

```

So, this way you can remove a lot of code duplicating. You will not need to search buttons, pass initial button texts and so on.

`handleSubmit` and `renderLoading` should be placed in `utils.js`, because they are utility functions